### ISSUE TITLE
M68K support?

### ISSUE THREAD

**Issue by lab313ru** [keystone-engine/keystone#312]
It would be great to support it (as in the Next branch of Capstone).

**Comment by lab313ru** [comment]
It has support for M68K too. http://xi6.com/projects/asmx/

**Comment by aquynh** [comment]
you are very much welcome to send a pull req to add M68K support to Keystone!

**Comment by shiftre** [comment]
Hehe, I'm a bit interested about it too, it looks like Unicorn supports it,(due to Qemu?), and there is
some unofficial [port](https://github.com/M680x0/M680x0-llvm) 

Is there any official documentation or some hints about adding new architectures?

**Comment by aquynh** [comment]
No docs, but i can say that adding a new architecture includes good amount of work.

I would make PPC a priority, if i have time to add a new architecture.

**Comment by aquynh** [comment]
Hmm i am confused: are you talking about adding a new arch to Unicorn (wrong place here)?

**Comment by shiftre** [comment]
it looks like unicorn supports [m68k](https://github.com/unicorn-engine/unicorn/search?utf8=%E2%9C%93&q=m68k&type=) but keystone doesn't.

There is a github repo with both [LLVM](https://github.com/M680x0/M680x0-llvm) and [Clang](https://github.com/M680x0/M680x0-clang) which supports it but I'm guessing it's not in master,
I don't mind adding support here, but I would need some guidance or at least some hints.

**Comment by shiftre** [comment]
What do you think? Merging this repo would help or not?

**Comment by aquynh** [comment]
yes definitely. to start, you can look at file https://github.com/keystone-engine/keystone/blob/master/llvm/keystone/ks.cpp to see how to add a new architecture.

**Comment by shiftre** [comment]
ok great.
